modules/java/native/java-grammar.ym

```diff
@@ -86,6 +87,14 @@ java_dest_custom_options
   |
   ;
 
+java_dest_empty_options
+  : java_dest_empty_option java_dest_empty_options
+  |
+  ;
+
+java_dest_empty_option
+  : LL_STRING { ;  }
+
```

Comment:
I included an example in my original post, but I will try to make it clearer here. 

If I have a java destination that optionally takes a parameter "foo", that is, behaves differently when the dictionary of parameters passed to Java includes "foo", with an associated value, I cannot write an SCL block for that destination currently without forcing "foo" to have a value always - that is, turning it from an optional option to a required option.

If in my scl block, I do not define a default value for "foo", parsing the block on startup will fail, because the grammar expects an option to have a value.

```
block destination test_block(foo()){
  java(
    class_name("sampleclass")
    class_path("/home/user/sampleclass/")
    option("foo",`foo`)
  );
};
#Syslog-ng would fail to parse this because a value wasn't given to foo
```
To appease the parser, I *could* include a default value for foo in the block definition, but that would mean "foo" will always be in the dictionary of options passed to Java - it no longer is an "optional" value.